sonic_db_config/db_config.go

```diff
@@ -1,135 +1,237 @@
+		DbInit()
+	}
+	db, ok := sonic_db_config[ns]["DATABASES"].(map[string]interface{})[db_name]
+	if !ok {
+		panic(fmt.Errorf("database name '%v' is not valid in database_config.json file for namespace `%v`!", db_name, ns))
+	}
+	inst_name, ok := db.(map[string]interface{})["instance"]
+	if !ok {
+		panic(fmt.Errorf("'instance' is not a valid field in database_config.json file for namespace `%v`!", ns))
+	}
+	inst, ok := sonic_db_config[ns]["INSTANCES"].(map[string]interface{})[inst_name.(string)]
+	if !ok {
+		panic(fmt.Errorf("instance name '%v' is not valid in database_config.json file for namespace `%v`!", inst_name, ns))
+	}
+	return inst.(map[string]interface{})
+}
+
+func GetDbSeparator(db_name string, ns string) string {
+	if !sonic_db_init {
+		DbInit()
+	}
+	db_list := GetDbList(ns)
+	separator, ok := db_list[db_name].(map[string]interface{})["separator"]
+	if !ok {
+		panic(fmt.Errorf("'separator' is not a valid field in database_config.json file!"))
+	}
+	return separator.(string)
+}
+
+func GetDbId(db_name string, ns string) int {
+	if !sonic_db_init {
+		DbInit()
+	}
+	db_list := GetDbList(ns)
+	id, ok := db_list[db_name].(map[string]interface{})["id"]
+	if !ok {
+		panic(fmt.Errorf("'id' is not a valid field in database_config.json file!"))
+	}
+	return int(id.(float64))
+}
+
+func GetDbSock(db_name string, ns string) string {
+	if !sonic_db_init {
+		DbInit()
+	}
+	inst := GetDbInst(db_name, ns)
+	unix_socket_path, ok := inst["unix_socket_path"]
+	if !ok {
+		panic(fmt.Errorf("'unix_socket_path' is not a valid field in database_config.json file!"))
+	}
+	return unix_socket_path.(string)
+}
+
+func GetDbHostName(db_name string, ns string) string {
+	if !sonic_db_init {
+		DbInit()
+	}
+	inst := GetDbInst(db_name, ns)
+	hostname, ok := inst["hostname"]
+	if !ok {
+		panic(fmt.Errorf("'hostname' is not a valid field in database_config.json file!"))
+	}
+	return hostname.(string)
+}
+
+func GetDbPort(db_name string, ns string) int {
+	if !sonic_db_init {
+		DbInit()
+	}
+	inst := GetDbInst(db_name, ns)
+	port, ok := inst["port"]
+	if !ok {
+		panic(fmt.Errorf("'port' is not a valid field in database_config.json file!"))
+	}
+	return int(port.(float64))
+}
+
+func GetDbTcpAddr(db_name string, ns string) string {
+	if !sonic_db_init {
+		DbInit()
+	}
+	hostname := GetDbHostName(db_name, ns)
+	port := GetDbPort(db_name, ns)
+	return hostname + ":" + strconv.Itoa(port)
+}
+
+func DbGetNamespaceAndConfigFile(ns_to_cfgfile_map map[string]string) {
+	data, err := io.ReadFile(SONIC_DB_GLOBAL_CONFIG_FILE)
+	if err == nil {
+		//Ref:https://stackoverflow.com/questions/18537257/how-to-get-the-directory-of-the-currently-running-file
+		dir, err := filepath.Abs(filepath.Dir(SONIC_DB_GLOBAL_CONFIG_FILE))
+		if err != nil {
+			panic(err)
+		}
+		var sonic_db_global_config = make(map[string]interface{})
+		err = json.Unmarshal([]byte(data), &sonic_db_global_config)
+		if err != nil {
+			panic(err)
+		}
+		var db_include_file string
```

Comment:
This var is not used outside the cycle and doesn't store any state, why does this var declared outside the for cycle?
what is the purpose of this var, it is used the next line after its initialization?

Reply:
updated.